tests/core/test_dogstatsd.py

```diff
@@ -1,892 +1,113 @@
-        from requests.utils import get_environ_proxies
-        from os import environ as env
-
-        env["http_proxy"] = "http://localhost:3128"
-        env["https_proxy"] = env["http_proxy"]
-        env["HTTP_PROXY"] = env["http_proxy"]
-        env["HTTPS_PROXY"] = env["http_proxy"]
-
-        self.assertTrue("no_proxy" in env)
-
-        self.assertEquals(env["no_proxy"], "127.0.0.1,localhost,169.254.169.254")
-        self.assertEquals({}, get_environ_proxies(
-            "http://localhost:17123/api/v1/series"))
-
-        expected_proxies = {
-            'http': 'http://localhost:3128',
-            'https': 'http://localhost:3128',
-            'no': '127.0.0.1,localhost,169.254.169.254'
-        }
-        environ_proxies = get_environ_proxies("https://www.google.com")
-        self.assertEquals(expected_proxies, environ_proxies,
-            (expected_proxies, environ_proxies))
-
-        # Clear the env variables set
-        del env["http_proxy"]
-        del env["https_proxy"]
-        if 'HTTP_PROXY' in env:
-            # on some platforms (e.g. Windows) env var names are case-insensitive, so we have to avoid
-            # deleting the same key twice
-            del env["HTTP_PROXY"]
-            del env["HTTPS_PROXY"]
+from dogstatsd import mapto_v6, get_socket_address
+from dogstatsd import Server
+from utils.net import IPV6_V6ONLY, IPPROTO_IPV6
+
+
+class TestFunctions(TestCase):
+    def test_mapto_v6(self):
+        self.assertIsNone(mapto_v6('foo'))
+        self.assertIsNone(mapto_v6('192.'))
+        self.assertEqual(mapto_v6('192.168.1.1'), '::ffff:192.168.1.1')
+        self.assertEqual(mapto_v6('::1'), '::1')
+        self.assertEqual(mapto_v6('ff00::'), 'ff00::')
+
+    def test_get_socket_address(self):
+        with mock.patch('dogstatsd.socket.getaddrinfo') as getaddrinfo:
+            getaddrinfo.return_value = [(2, 2, 17, '', ('192.168.1.1', 80))]
+            self.assertEqual(get_socket_address('example.com', 80), ('::ffff:192.168.1.1', 80, 0, 0))
+            getaddrinfo.return_value = [(30, 2, 17, '', ('::1', 80, 0, 0))]
+            self.assertEqual(get_socket_address('example.com', 80), ('::1', 80, 0, 0))
+        self.assertIsNone(get_socket_address('foo', 80))
+
+
+class TestServer(TestCase):
+    @mock.patch('dogstatsd.get_socket_address')
+    def test_init(self, nh):
+        nh.return_value = 'foo'
+
+        s = Server(None, 'localhost', '1234')
+        nh.assertCalledOnceWith('localhost', 1234)
+
+        self.assertEqual(s.sockaddr, 'foo')
+        self.assertIsNone(s.socket)
+
+    @mock.patch('dogstatsd.select')
+    def test_start(self, select):
+        select.select.side_effect = [KeyboardInterrupt, SystemExit]
+        s1 = Server(mock.MagicMock(), '::1', '1234')
+        s1.start()
+        self.assertEqual(s1.socket.family, socket.AF_INET6)
+
+        s2 = Server(mock.MagicMock(), '127.0.0.1', '2345')
+        s2.start()
+        self.assertEqual(s2.socket.family, socket.AF_INET6)
```

Comment:
@masci not sure if this is because of the MagicMock()... shouldn't this be `socket.AF_INET`?